src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionsSource.cs

```diff
@@ -162,15 +183,18 @@ public bool TryGetTelemetryId(out Guid telemetryId)
 
                 if (_workspaceStatusService != null)
                 {
-                    // TODO: right now, LightBulb uses IVsThreadedWaitDialog directly rather than new
-                    //       IUIThreadOperationContext. we need to talk to editor team whether we want to
-                    //       update API to pass in OperationContext like any new API, or we use 
-                    //       IWaitIndicator abstraction (which is a thin wrapper on top of IVsThreadedWaitDialog) directly
-                    //       for now, we use the one LB created before calling us. meaning we don't update
-                    //       text on the wait dialog window
-                    // 
-                    //       this also needs to run under threading context otherwise, we can deadlock on VS
-                    ThreadingContext.JoinableTaskFactory.Run(() => _workspaceStatusService.WaitUntilFullyLoadedAsync(cancellationToken));
+                    IDisposable? operationScope = null;
+                    try
+                    {
+                        operationScope = addOperationScope?.Invoke("WorkspaceStatusService");
+
+                        // This needs to run under threading context otherwise, we can deadlock on VS
+                        ThreadingContext.JoinableTaskFactory.Run(() => _workspaceStatusService.WaitUntilFullyLoadedAsync(cancellationToken));
```

Comment:
this is terrifying.

Reply:
This is not new code. Please feel free to open a new issue if you see areas for improvement though.